kotlin/src/test/kotlin/org/jdbi/v3/core/kotlin/KotlinMapperTest.kt

```diff
@@ -40,6 +40,22 @@ class KotlinMapperTest {
 
     }
 
+    @Test fun shouldUseSecondOccurrenceOfColumnWhenMultipleColumnsWithSameNameArePresent() {
+
+        val mapper = KotlinMapper(DataClassWithOnlyPrimaryConstructor::class.java)
+        mockColumns("id", "name", "name")
```

Comment:
Now that I look at this, I'm not sure I agree with this use case.

This appears to hardcode an assumption that if a result set has two columns with the same name, we always want to use the latter. That may work in some scenarios but I don't think it's something we should encourage.

I rather think it's better to leave which column will be selected undefined, and recommend that users formulate their queries to be unambiguous.

I'm open to hear any arguments you have in favor of this, but my gut reaction is that this doesn't seem like the kind of decision we should unilaterally make for our users.